File: app/src/main/java/com/example/ep2024/domain/model/Role.java
package com.example.ep2024.domain.model;

import androidx.annotation.NonNull;

public enum Role {
    ADMIN("ADMIN"),
    USER("USER"),
    PUP("PUP"),
    OD("OD");

    private final String _str;

    Role(String str) {
        _str = str;
    }

    @NonNull
    @Override
    public String toString() {
        return _str;
    }

    public static Role fromString(String str){
        for (Role role : Role.values()) {
            if (role._str.equals(str)) {
                return role;
            }
        }
        return null;
    }

}

File: app/src/main/java/com/example/ep2024/domain/model/Role.kt
package com.example.ep2024.domain.model

enum class Role(private val _str: String) {
    ADMIN("ADMIN"),
    USER("USER"),
    PUP("PUP"),
    OD("OD");

    override fun toString(): String {
        return _str
    }

    companion object {
        fun fromString(str: String): Role? {
            for (role in entries) {
                if (role._str == str) {
                    return role
                }
            }
            return null
        }

        @JvmStatic
        val rolesListWithoutAdminRole: List<Role>
            get() = ArrayList(
                listOf(
                    USER,
                    OD,
                    PUP
                )
            )

        val rolesList: List<Role>
            get() = ArrayList(
                listOf(
                    ADMIN,
                    USER,
                    OD,
                    PUP
                )
            )
    }
}
